--- include/octomap/OcTreeBase.h	2011-08-17 03:11:21.000000000 -0500
+++ include/octomap/OcTreeBase.h	2011-11-06 05:47:23.000000000 -0600
@@ -355,7 +355,7 @@
         if (top.depth == maxDepth)
           return;
 
-        iterator_base::StackElement s;
+        StackElement s;
         s.depth = top.depth +1;
 
         unsigned short int center_offset_key = tree->tree_max_val >> (top.depth +1);